cmd/govim/testdata/scenario_default/workspace_symbol.txt

```diff
@@ -0,0 +1,39 @@
+# Test that GOVIMWorkspaceSymbol works as expected
+
+vim -indent expr 'GOVIMWorkspaceSymbol({\"query\": \"main\"})'
```

Comment:
Is govim's philosophy generally to pass through the request parameters like this? I think it would be nicer to have GOVIMWorkspaceSymbol accept a string.

Reply:
> Is govim's philosophy...

I don't think I/we have ever had a philosophy ;-)

Joking aside, this is the first `gopls`/LSP function we have exposed directly to Vim with no intermediate step. Part of the reason for simply acting as a pass-through is that otherwise we need to establish a process for flattening the params struct:

https://github.com/golang/tools/blob/ea829e2eb296fe30ae77de319c26e88e3b165652/internal/lsp/protocol/tsprotocol.go#L3525-L3533

working out what types to use (for example `WorkDoneToken` would likely need to be a `*interface{}` to avoid the zero-value for a `string`/`int` meaning something) and then documenting those parameters. 

So in many respects I was being lazy.

Ideas very much welcomed for alternative approaches.